monai/apps/pathology/transforms/spatial/array.py

```diff
@@ -73,3 +75,133 @@ def get_params(self, image_size):
         )
 
         return patch_size, steps
+
+
+class TileOnGrid(Transform):
+    """
+    Tile the image into patches on a grid and maintain a subset of it.
+    This transform works only with np.ndarray inputs.
+
+    Args:
+        tile_count: number of tiles to extract
+        tile_size: size of the square tile
+            Defaults to ``256``.
+        tile_step: step size
+            Defaults to None (same as tile_size)
+        tile_all: Extract all non-background tiles, instead of tile_count.
+            Defaults to ``False``.
+        tile_random_offset: Randomize position of tile grid, instead of starting from the top-left corner
+            Defaults to ``False``.
+        tile_pad_full: pad image to the size evenly divisible by tile_size
+            Defaults to ``False``.
+        tile_background_val: the background constant (e.g. 255 for white background)
+            Defaults to ``255``.
+        tile_filter_mode: [min, max, None] filter tiles. If total number of tiles is more then tile_size,
+            then sort by intensity sum, and take the smallest (for min), largest (for max) or random (for None) subset
+            Defaults to ``min`` (which assumes background is white, high value)
+        random_seed: provide random seed int manually
+            Defaults to None
+
+    """
+
+    def __init__(
+        self,
+        tile_count: int = None,
+        tile_size: int = 256,
+        tile_step: int = None,
+        tile_all: bool = False,
+        tile_random_offset: bool = False,
+        tile_pad_full: bool = False,
+        tile_background_val: int = 255,
+        tile_filter_mode: str = "min",  # None, min, max
+        random_seed: int = None,
```

Comment:
If your transform is random transform, please extends from MONAI `Randomizable` and do random logic in the `randomize()` function:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/transform.py#L122
I think you can refer to `RandSpatialCrop`:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/croppad/array.py#L479

Thanks.

Reply:
added Randomizable, and changed unit tests